src/js/control-bar/progress-control/seek-bar.js

```diff
@@ -39,7 +40,13 @@ class SeekBar extends Slider {
    *        The key/value store of player options.
    */
   constructor(player, options) {
-    super(player, options);
+    const options_ = merge({
+      playerOptions: {
+        enableSmoothSeeking: false
+      }
+    }, options);
+
+    super(player, options_);
```

Comment:
while it makes sense for this to be a player option, I think that rather than setting a default player option here, we should have it be translated into a seek-bar option called `enableSmoothSeeking`.

Maybe something like this (I didn't verify this works)

```suggestion
    const opts = merge({
      enableSmoothSeeking: options.playerOptions.enableSmoothSeeking ?? false
    }, options);

    super(player, opts);
```

Then below, it could use `this.options_.enableSmoothSeeking`

Reply:
I guess the alternative would be to always do `this.player().options().enableSmoothSeeking` (then someone could more easily update this on the fly and won't need to drill down to the seek bar via `player.controlBar.progressControl.seekBar.options_.playerOptions.enableSmoothSeeking = true`, which I had to do to test this out. However, I think making it a seek-bar option is fine and makes sense.

Reply:
@gkatsev thanks for the suggestion. 

> I guess the alternative would be to always do `this.player().options().enableSmoothSeeking `(then someone could more easily update this on the fly and won't need to drill down to the seek bar

Your suggestion is much better because it offers a nicer developer experience. 

Here's what the changes could look like:

**seek-bar.js**
```javascript
  constructor(player, options) {
    super(player, options);
    this.setEventHandlers_();
  }

  handleMouseMove(event, mouseDown = false) {
     // ...
     if (this.player_.options_.enableSmoothSeeking) {
      this.update();
    }
  }
```

**time-display.js**

```javascript
  constructor(player, options) {
    super(player, options);

    this.on(player, ['timeupdate', 'ended', 'seeking'], (e) => this.update(e));
    this.updateTextNode_();
  }

  update(event) {
    if (!this.player_.options_.enableSmoothSeeking && event.type === 'seeking') {
      return;
    }

    this.updateContent(event);
  }
```

These changes remove `merge` `options`, reducing the number of lines of code to maintain. As a result, a developer will be able to toggle the smooth seeking on the fly:


```javascript
player.options({enableSmoothSeeking : true});
player.options({enableSmoothSeeking : false});
```

Finally, relying directly on player options is something already done for other components: `clickable-component`, `live-tracker`, `poster-image` and `picture-in-picture-toggle`.

What do you think about these changes?

Reply:
Sounds good to me!

Reply:
@mister-ben thank you for the feedback, I'll try to make the changes by the end of the week.

Reply:
I pushed the changes. However, I don't see where I can add the documentation for this new option.